### ISSUE TITLE
PHPDoc tag @var for constant with type string|null is not subtype of native type string

### ISSUE THREAD

**Issue by gabrielrbarbosa** [nunomaduro/larastan#1928]
- Larastan Version: 11.x
- Laravel Version: 5

### Description
Errors detected in CI

```
16     PHPDoc tag @var for constant                                
         App\Models\Produto\CustoBanhoProduto::CREATED_AT with type  
         string|null is not subtype of native type string.
           
17     PHPDoc tag @var for constant                                
         App\Models\Produto\CustoBanhoProduto::UPDATED_AT with type  
         string|null is not subtype of native type string. 
```

### Laravel code where the issue was found

```php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class TestModel extends Model
{
    use HasFactory;

    protected $table = 'test';

    public $timestamps = true;
    const string CREATED_AT = 'data_criacao';
    const string UPDATED_AT = 'data_alteracao';
    
    ...
}
```

**Comment by calebdw** [comment]
Update your type to `?string`

**Comment by szepeviktor** [comment]
> Update your type to `?string`

Constants will never change type.

**Comment by calebdw** [comment]
The `@var` tags come from the framework---and you can override the constant to be `null` on a particular model.

There's nothing for us to do here---you either need to remove the type or set to `?string`.

**Comment by gabrielrbarbosa** [comment]
I think the expected behavior is that a `const string` should be valid for "string|null" right?

**Comment by calebdw** [comment]
I would say no, but that's something you can take up with the PHPStan side of things.

**Comment by szepeviktor** [comment]
https://github.com/laravel/framework/blob/22f4864fcb54cd04fc0e110af193aa9a63905e22/src/Illuminate/Database/Eloquent/Model.php#L214-L219

**Comment by canvural** [comment]
Hi,

This doesn't seem related to Larastan. Please produce it on PHPStan playground and report it there. Thanks.

**Comment by gabrielrbarbosa** [comment]
phpstan fixed it! waiting for new 2.1.x release
